### PR TITLE
Feature/accept serialized binary array

### PR DESCRIPTION
**Pull request by scalarindetail**
This PR implements feature #16. There are several things to review:

- The new class `ClientServiceWithBinary`
JavaScript doesn't support function overloading.
To make the function names synchronized with Java SDK.
We create a new class ClientServiceWithBinary and implement all binary related functions in this class.

- The response of these functions
We just return with native gRPC response objects.

- The integration test
Instead of extending `test/integration_test.js`, we create another file to test new functions because it is a bit hard to achieve idempotence in the same file.

- README
The new section to explain how to use ClientServiceWithBinary.

### CODE REVIEW

File: scalardl-node-client-sdk.js [hunk 1]
```diff
@@ -17,32 +17,83 @@ class ClientService extends ClientServiceBase {
    */
   constructor(properties) {
     const ledgerClientUrl =
-      `${properties['scalar.dl.client.server.host']}:` +
-      `${properties['scalar.dl.client.server.port']}`;
+        `${properties['scalar.dl.client.server.host']}:` +
+        `${properties['scalar.dl.client.server.port']}`;
```
Comment (scalarindetail) [inline scalardl-node-client-sdk.js]: I'm not sure why google javascript will add indentation here, is it the same for you as well Plenty?

Comment (supl) [reply scalardl-node-client-sdk.js]: I guess it is made by WebStorm.
Google JavaScript style guide will do indent with 2 spaces
https://google.github.io/styleguide/jsguide.html#formatting-block-indentation

File: scalardl-node-client-sdk.js [hunk 2]
```diff
@@ -27,23 +27,174 @@ class ClientService extends ClientServiceBase {
...
     }
     const services = {
       'ledgerClient': ledgerClient,
       'ledgerPrivileged': ledgerPrivilegedClient,
     };
     super(services, protobuf, properties);
   }
+
+  /**
+   * @param {Uint8Array} serializedBinary
```
Comment (feeblefakie) [inline scalardl-node-client-sdk.js]:
```suggestion
   * @param {Uint8Array} serializedBinary a serialized byte array of CertificateRegistrationRequest
```
I'm not sure if we can hav a comment like this in nodejs though.

Comment (scalarindetail) [reply scalardl-node-client-sdk.js]: No problem.
It is a correct JSDoc.

Comment (supl) [reply scalardl-node-client-sdk.js]: No problem.
It's a correct JSDoc.

**Comment by supl** [comment]
> @supl
> Sorry for late. Let's change the name of class `ClientService` to `ClientServiceWithBinary` and use the same named methods.

@feeblefakie 
Patched.
Please kindly review :)

File: README.md [hunk 3]
```diff
@@ -140,6 +140,106 @@ StatusCode = {
 };
 ```
 
+## Create raw gRPC requests
+
+Users can use SDK to create raw gRPC requests in byte arries (JavaScript Uint8Array) too.
```
Comment (feeblefakie) [inline README.md]:
```suggestion
You can also create a raw gRPC request in byte array (JavaScript Uint8Array) too.
```

File: README.md [hunk 4]
```diff
@@ -140,6 +140,106 @@ StatusCode = {
...
+The functions have different names but the parameters are exactly identical.
```
Comment (feeblefakie) [inline README.md]:
```suggestion
Note that the name of functions are different from usual functions such as `executeContract` but the parameters are exactly the same.
```

File: README.md [hunk 5]
```diff
@@ -140,6 +140,106 @@ StatusCode = {
...
+const binary = await clientService.createSerializedContractExecutionRequest('contractId', argumentObject);
+```
+
+### Validate an asset
+```javascript
+const binary = await clientService.createSerializedLedgerValidationRequest('assetId');
+```
+
+## Send the raw gRPC requests to Scalar DL servers
+The SDK provides another `ClientServiceWithBinary` class whereby users can send byte arries of requests to Scalar DL servers.
```
Comment (feeblefakie) [inline README.md]:
```suggestion
The SDK has another `ClientServiceWithBinary` class for you to send the byte array of a request to Scalar DL network.
```

File: README.md [hunk 6]
```diff
@@ -140,6 +140,106 @@ StatusCode = {
...
+It is similar with `ClientService` class to construct `ClientServiceWithBinary` class.
+However, ClientServiceWithBinary is different from ClientService.
+ClientService encapsulates responses into Scalar defined objects but ClientServiceWithBinary returns gRPC native objects.
```
Comment (feeblefakie) [inline README.md]: Oh ,I think they should return the same object even if serialized byte array is an input.

**Comment by supl** [comment]
> Overall looking good but I think the methods of `ClientServiceWithBinary` should return the same objects as `ClientService`

@feeblefakie 
Thanks for your feedback.
I revised again according to the comments.
Please review again :)

File: README.md [hunk 7]
```diff
@@ -140,6 +140,91 @@ StatusCode = {
...
+```
+
+### Validate an asset
+```javascript
+const binary = await clientService.createSerializedLedgerValidationRequest('assetId');
+```
+
+## Send the raw gRPC requests to Scalar DL servers
+The SDK has another `ClientServiceWithBinary` class for you to send the byte array of a request to Scalar DL network.
+It is similar with `ClientService` class to construct `ClientServiceWithBinary` class.
```
Comment (feeblefakie) [inline README.md]: I think we can delete this line without losing ease of understanding. :)

Comment (supl) [reply README.md]: @feeblefakie 
Revised.
By the way, could you publish SDK base version 2.4.5 (which fixed the issue of optional contract arguments)?
Then, I can update it in the PR as well :)
Thank you.

Comment (feeblefakie) [reply README.md]: Sorry, it is just published !

Comment (supl) [reply README.md]: > Sorry, it is just published !

@feeblefakie 
Thank you.
Updated.
Please review again :)
Sorry for taking so long.

**Pull request merged**